docs/porting/standard_pin_names/pin_names_porting.md

```diff
@@ -0,0 +1,234 @@
+# Standard Pin Names Porting Guide
+
+This guide aims to help vendors to comply with the new pin naming guidelines.
+
+There is a set of generic guidelines that apply to all boards, and further sets of guidelines that additionally apply to boards that have certain compatible connectors. The current guideline documents can be found here:
+* [General Pin Names Design Document](https://github.com/ARMmbed/mbed-os/blob/feature-std-pin-name/docs/design-documents/hal/0004-pin-names-general-guidelines.md)
```

Comment:
Let's remove links to design docs, as we'll remove them from Mbed OS to this porting guide and developer guide (WIP)

Reply:
So all information from the design docs have to be copied here ?

Reply:
The info from design docs is copied to either the porting guide (here) or the developer guide (pending to be added).